src/librustc/ty/maps.rs

```diff
@@ -96,9 +96,9 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-pub struct CycleError<'a, 'tcx: 'a> {
+pub struct CycleError<'tcx> {
     span: Span,
-    cycle: RefMut<'a, [(Span, Query<'tcx>)]>,
+    cycle: Vec<(Span, Query<'tcx>)>,
```

Comment:
> Why do you want to query from the printing code?

I don't *necessarily* want that. But the printing code currently pokes at the query's internal maps to find out if `trait_ref` and `ty` are filled out for particular def-ids (or something like that). I guess I could add an accessor for reading the current state of the query that you are *really, really* not supposed to use (because it lets you observe things you shouldn't be able to observe), just for use in printing things out -- but really it seems like it might make sense for printing things out to run bits of type-check that may not have run, if it is needed to pretty-print better (i.e., if we were in a totally on-demand scenario, and some error arose, it seems plausible it might need to pretty print something that hasn't yet been "collected"--- well, maybe that would never happen in practice).